core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/machine/Machine.java

```diff
@@ -14,10 +14,43 @@
  * Defines runtime machine.
  *
  * @author gazarenkov
+ * @author Alexander Garagatyi
  */
-public interface Machine extends MachineState {
+public interface Machine {
     /**
-     * Returns machine specific metadata
+     * Returns configuration used to create this machine
      */
-    MachineMetadata getMetadata();
+    MachineConfig getConfig();
+
+    /**
+     * Returns machine identifier. It is unique and mandatory.
+     */
+    String getId();
+
+    /**
+     * Returns ID of workspace this machine belongs to. It is mandatory.
+     */
+    String getWorkspaceId();
+
+    /**
+     * Returns name of environment that started this machine. It is mandatory.
+     */
+    String getEnvName();
+
+    /**
+     * Returns machine owner (users identifier). It is mandatory.
+     */
+    String getOwner();
+
+    /**
+     * Runtime status of the machine
+     */
+    MachineStatus getStatus();
+
+    /**
+     * Runtime information about machine.
+     * <p>
+     * Is available only when {@link #getStatus()} returns {@link MachineStatus#RUNNING}
+     */
+    MachineRuntimeInfo getRuntime();
```

Comment:
Move docker properties (which currently are in MachineMetadata) directly to MachineRuntimeInfo
So remove MachineMetadata